### PR TITLE
Update gQSPsim with contributions from MathWorks

### PR DESCRIPTION
**Pull request by simkaryote**
THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY.

### CODE REVIEW

**Review by feigelman — COMMENTED** [review]
removing previous comment

**Comment by feigelman** [comment]
> running the new version of the code locally after merging I immediately encounter problems with the advanced logger. That directory app/+QSPViewerNew/+Widgets/AdvLogger is empty. I assume that is supposed to track https://github.com/mathworks/advanced-logger as a submodule. However, even after cloning that in I encounter the following error. Do you have a modified local version of this module? It's not in the repository that I can tell. Otherwise I don't know why it's looking for undefined properties in the Logger class.
> 
> No public property 'FileID' for class 'Logger'.
> 
> Error in QSPViewerNew.Widgets.Logger/fcloseLogFile (line 198) if obj.FileID >= 0
> 
> Error in QSPViewerNew.Widgets.Logger/moveLogFile (line 93) obj.fcloseLogFile();
> 
> Error in QSP.Session/updateLoggerName (line 847) moveLogFile(loggerObj, obj.RootDirectory); % moves log file if not present in root dir
> 
> Error in QSP.Session/setSessionName (line 544) updateLoggerName(obj, SessionName)
> 
> Error in QSPViewer.App/refresh (line 81) setSessionName(obj.Session(idx),ThisRawName);
> 
> Error in uix.abstract.AppWithSessionFiles/createUntitledSession (line 223) obj.refresh(); %Call refresh of the main app
> 
> Error in uix.abstract.AppWithSessionFiles/onNew (line 153) createUntitledSession(obj);
> 
> Error in uix.abstract.AppWithSessionFiles>@(h,e)onNew(obj) (line 66) 'Callback',@(h,e)onNew(obj));
> 
> Error while evaluating Menu Callback.

With this error, I cannot launch the app

**Comment by simkaryote** [comment]
See

https://stackoverflow.com/questions/3796927/how-do-i-git-clone-a-repo-including-its-submodules

Pax
________________________________

**Review by feigelman — DISMISSED** [review]

File: app/+QSP/+abstract/BaseProps.m
Comment (feigelman) [inline app/+QSP/+abstract/BaseProps.m:75]: not sure we need Log anymore since we have the Logger class. This probably needs to be removed.

File: app/+QSP/+abstract/BasicBaseProps.m
Comment (feigelman) [inline app/+QSP/+abstract/BasicBaseProps.m:61]: Do we want to externalize version numbering? Otherwise if there are other versions across the code base we will have to maintain consistency for updates.

File: app/+QSP/@CohortGeneration/CohortGeneration.m
Comment (feigelman) [inline app/+QSP/@CohortGeneration/CohortGeneration.m:37]: I think ultimately we wanted to use the full strings in the UI. It doesn't matter if that is represented as a shortened string in the backend.

File: app/+QSP/@CohortGeneration/CohortGeneration.m
Comment (feigelman) [inline app/+QSP/@CohortGeneration/CohortGeneration.m:601]: probably need to remove this since we are using a new logger

File: app/+QSP/@CohortGeneration/CohortGeneration.m
Comment (feigelman) [inline app/+QSP/@CohortGeneration/CohortGeneration.m:623]: will need to revisit the whole SQL database aspect

File: app/+QSP/@Optimization/Optimization.m
Comment (feigelman) [inline app/+QSP/@Optimization/Optimization.m:712]: switch to new logger

File: app/+QSP/@Optimization/Optimization.m
Comment (feigelman) [inline app/+QSP/@Optimization/Optimization.m:727]: revisit SQL

File: app/+QSP/@Settings/refreshData.m
Comment (feigelman) [inline app/+QSP/@Settings/refreshData.m:17]: seems like this will work for now. in the future let's move all the waitbars into the viewer?

File: app/+QSP/@Settings/refreshData.m
Comment (feigelman) [inline app/+QSP/@Settings/refreshData.m:34]: same here

File: app/+QSP/@Simulation/Simulation.m
Comment (feigelman) [inline app/+QSP/@Simulation/Simulation.m:326]: update to Logger

File: app/+QSP/@Simulation/Simulation.m
Comment (feigelman) [inline app/+QSP/@Simulation/Simulation.m:347]: revisit

File: app/+QSP/@VirtualPopulationGeneration/VirtualPopulationGeneration.m
Comment (feigelman) [inline app/+QSP/@VirtualPopulationGeneration/VirtualPopulationGeneration.m:520]: update

File: app/+QSP/@VirtualPopulationGeneration/VirtualPopulationGeneration.m
Comment (feigelman) [inline app/+QSP/@VirtualPopulationGeneration/VirtualPopulationGeneration.m:540]: revisit

File: app/+QSP/plotVirtualCohortGeneration.m
Comment (feigelman) [inline app/+QSP/plotVirtualCohortGeneration.m:630]: need to revisit

File: app/+QSPViewer/@App/App.m
Comment (feigelman) [inline app/+QSPViewer/@App/App.m:536]: update

File: app/+QSPViewer/@App/App.m
Comment (feigelman) [inline app/+QSPViewer/@App/App.m:627]: update

File: app/+QSPViewer/@App/loadSessionFromFile.m
Comment (feigelman) [inline app/+QSPViewer/@App/loadSessionFromFile.m:63]: why was this removed?

File: app/+QSPViewer/@Optimization/Optimization.m
Comment (feigelman) [inline app/+QSPViewer/@Optimization/Optimization.m:111]: why just a warning?

File: app/+QSPViewer/@Session/create.m
Comment (feigelman) [inline app/+QSPViewer/@Session/create.m:105]: we need to clean up the logger section. Since we're using the advanced logger, don't need these properties anymore?

File: app/+QSPViewer/@Session/create.m
Comment (feigelman) [inline app/+QSPViewer/@Session/create.m:172]: revisit if we are leaving out SQL capability

File: app/+QSPViewer/@Session/create.m
Comment (feigelman) [inline app/+QSPViewer/@Session/create.m:232]: delete commented lines

File: app/+QSPViewer/@Session/refresh.m
Comment (feigelman) [inline app/+QSPViewer/@Session/refresh.m:38]: update/review

File: app/+QSPViewer/@Session/update.m
Comment (feigelman) [inline app/+QSPViewer/@Session/update.m:46]: update/review

**Review by feigelman — COMMENTED** [review]
generally we should remove unneeded .mat files, .xlsx files, old code from +QSPViewer.

Marked a few places where we need to clean up old vs new logger paradigm, and remove/update calls to store results in the SQLite DB. I did not carefully review the new classes in the QSPViewer, and instead focused on changes to the backend and other/utility code.

File: app/+QSP/@Settings/refreshData.m
Comment (simkaryote) [reply app/+QSP/@Settings/refreshData.m:17]: yes, all UI elements (including waitbars should be in the viewer).

File: app/+QSP/@CohortGeneration/CohortGeneration.m
Comment (simkaryote) [reply app/+QSP/@CohortGeneration/CohortGeneration.m:601]: Yes, but that should be a project independently done from the refactor. It is best to try to keep changes atomic for various reasons, one of which is enabling the possibility of reverting or editing a particular change(s).

File: app/+QSP/@Optimization/Optimization.m
Comment (simkaryote) [reply app/+QSP/@Optimization/Optimization.m:712]: sure.. fit with above comment.

File: app/+QSPViewer/@Session/create.m
Comment (simkaryote) [reply app/+QSPViewer/@Session/create.m:105]: Probably not.. again that is another project unrelated to the refactor.

File: app/+QSPViewer/@Session/create.m
Comment (simkaryote) [reply app/+QSPViewer/@Session/create.m:172]: that would be your call. My suggestion would be to leave the code and disable it via some feature control mechanism. Simplest form of that would be to add a table of boolean flags to the corresponding classes (in this case the view) that conditionally expose functionality. In the simplest form this can be done by removing UI elements only and not touching any backend code.

File: app/+QSPViewer/@Session/create.m
Comment (simkaryote) [reply app/+QSPViewer/@Session/create.m:232]: In general UIX code should go away. The code is probably close to that.

File: app/+QSPViewer/@Session/refresh.m
Comment (simkaryote) [reply app/+QSPViewer/@Session/refresh.m:38]: I would prefer to not include new code in the refactor. In particular as this code is not written by MathWorks and it's going in with a pull request casting the code changes as MathWorks contributions. I am going to look the other way in this case.  But my recommendation going forward for your own sanity.

File: app/+QSPViewer/@Session/update.m
Comment (simkaryote) [reply app/+QSPViewer/@Session/update.m:46]: same comment as above.

**Comment by simkaryote** [comment]
I am gong to force the pull request into dev. It's been too long waiting for the approval and code changes from outside MathWorks are leaking into this pull request. I don't think it is controversial anymore that this new refactor will be part of v2.

**Pull request merged**
